### PR TITLE
Add canvas STORAGE_BINDING cases for RGBA8Unorm and RGBA16Float

### PR DESCRIPTION
**Pull request by jchen10**
Issue: #915 #916 

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

**Comment by jchen10** [comment]
@kainino0x @austinEng @Kangz PTAL

**Comment by kainino0x** [comment]
Please request reviews using the GitHub "Reviewers" field (top right of the pull request page) so this gets routed correctly by email filters :)

**Review by kainino0x — APPROVED** [review]

File: src/webgpu/web_platform/reftests/canvas_complex.html.ts [hunk 1]
```diff
@@ -65,6 +69,63 @@ export function run(
             data.set(new Uint8Array([0x7f, 0x7f, 0x00, 0xff]), 256 + 4); // yellow
           }
           break;
+        case 'rgba16float':
+          {
+            // ref: http://stackoverflow.com/questions/32633585/how-do-you-convert-to-half-floats-in-javascript
```
[the remark this answers is not in the document]
Comment (jchen10) [reply src/webgpu/web_platform/reftests/canvas_complex.html.ts]: Thanks, glad to know it.

File: src/webgpu/web_platform/reftests/canvas_complex.html.ts [hunk 2]
```diff
@@ -21,6 +24,7 @@ export function run(
     switch (format) {
       case 'bgra8unorm':
       case 'rgba8unorm':
+      case 'rgba16float':
```
[the remark this answers is not in the document]
Comment (jchen10) [reply src/webgpu/web_platform/reftests/canvas_complex.html.ts:27]: Done!

**Comment by jchen10** [comment]
> Please request reviews using the GitHub "Reviewers" field (top right of the pull request page) so this gets routed correctly by email filters :)

Done, sorry for my bad memory!

**Comment by jchen10** [comment]
@kainino0x  BTW, I am having the permission problem with "git push" to my jie-DryRun branch. As a result, I can't know the bots status about this PR ahead of the actual roll. That may bring some extra efforts for the unfortunate roller who will have to clean up the expectation for the PR. Basically I would only use the push for my dryrun branch. I am okay with the either way as long as convenient for you.

**Pull request merged**